### PR TITLE
Fix hardware selector picking with many cells

### PR DESCRIPTION
**Pull request by bruyeret**
Picking with >2**24 cells was broken as the `ID_HIGH24` didn't work.
It is likely that it never worked.
Add a test to avoid regressions.

### CODE REVIEW

**Comment by bruyeret** [comment]
I don't understand why the test is failing in Firefox
I tried with Firefox 128.0.3 on Windows 11 and Firefox 129.0 in WSL (Ubuntu 22.04) and it works fine
Maybe the error is specific to Firefox 128 on Ubuntu

image
In WSL

image
In Windows 11

File: Sources/Rendering/Core/HardwareSelector/test/testHardwareSelectorManyCells.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  const renderWindow = gc.registerResource(vtkRenderWindow.newInstance());
+  const renderer = gc.registerResource(vtkRenderer.newInstance());
+  renderWindow.addRenderer(renderer);
+  renderer.setBackground(0.32, 0.34, 0.43);
+
+  // Many cells to force high24 pass
+  const bigPolydata = gc.registerResource(vtkPolyData.newInstance());
+  const numberOfTriangles = 17000000;
+  const pointsData = Float32Array.from([
+    // Triangle points
```
Comment (finetjul) [inline Sources/Rendering/Core/HardwareSelector/test/testHardwareSelectorManyCells.js:38]: // Points for the last triangle

File: Sources/Rendering/OpenGL/PolyDataMapper/index.js [hunk 2]
```diff
@@ -1761,6 +1761,8 @@ function vtkOpenGLPolyDataMapper(publicAPI, model) {
     if (publicAPI.getNeedToRebuildBufferObjects(ren, actor)) {
       publicAPI.buildBufferObjects(ren, actor);
     }
+    // Always call this function as the selector can change
+    publicAPI.updateMaximumPointCellIds();
```
Comment (finetjul) [inline Sources/Rendering/OpenGL/PolyDataMapper/index.js:1765]: please explain in commit description why you moved `updateMaximumPointCellIds()`

**Review by finetjul — APPROVED** [review]
LGTM

**Review by sankhesh — APPROVED** [review]
:+1:

**Pull request merged**